### PR TITLE
Add Back button

### LINKED ISSUE #277: Add Back button to Vital Records flow
### Background 
To better align with CDT/CA State Design norms, we would like to add "Back" and "Next" Buttons to the application flow. We should do this before we expand the flow to include options for all 3 records types: birth, marriage and death. 

See [Figma screens](https://www.figma.com/design/CRJFJW74ZZ24mrEPS4YCmx/Vital-Records-Flow?node-id=2406-1087&p=f&t=GT1CPBY8CMTjP5iC-0) for design updates. 

### TODOs 

- [ ] Add support for back and next buttons to flow 
- [ ] Test functionality to ensure all edge cases and use cases are covered

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #277 

## What this PR does
- Creates an includes called Button Row that will add a Back/Next. The Next label is customizable. 
- [x] Add a back button on the `form.html` template, wired up as an `a href` with role of `button` that routes to `previous_route`.
- [x] Adds back button includes manually to sworn statement, type, order and confirm pages. The select fire page does not have a Back button.

### CODE REVIEW

**Comment by machikoyasuda** [comment]
@lalver1 @angela-tran When y'all run this locally, is the back button actually taking you back? Mine's taking me to the route it's currently on.

**Comment by machikoyasuda** [comment]
@angela-tran I noticed the same thing! The debugger paused during the `views` method _is_ in the correct state. But then it somehow.... changes...

File: web/vital_records/templates/vital_records/_button_row.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="btn-row">
+    <!-- djlint:off -->
+      <a class="btn btn-outline-primary"
+         role="button"
+         href="{% url previous_route pk=vital_records_request.id as previous_url %}">Back</a><button type="submit" class="btn btn-primary">{{ submit_label }}</button>
```
[the remark this answers is not in the document]
Comment (angela-tran) [reply web/vital_records/templates/vital_records/_button_row.html]: The `SubmitView` sets [`context_object_name = "vital_records_request"`](https://github.com/Office-of-Digital-Services/cdt-ods-disaster-recovery/blob/main/web/vital_records/views.py#L275), and that's how its template is able to use the object as [`vital_records_request`](https://github.com/Office-of-Digital-Services/cdt-ods-disaster-recovery/blob/main/web/vital_records/templates/vital_records/request/confirm.html#L14).

[the remark this answers is not in the document]
Comment (angela-tran) [reply web/vital_records/templates/vital_records/_button_row.html]: I see you went with `object`, which works! :+1:

File: web/vital_records/templates/vital_records/_button_row.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="btn-row">
+    <a class="btn btn-outline-primary"
+       role="button"
+       href="{{ previous_url }}">Back</a>
```
Comment (machikoyasuda) [inline web/vital_records/templates/vital_records/_button_row.html:4]: Originally, the `href` line was being set with: `{% url previous_route pk=object.id %}` - a URL being created from the supplied `previous_route` from the context, and the ID key being passed in.

This was causing a bug on the Type form view, where the `previous_route` is going back to `request_start`, which does not need the ID in the URL. 

Instead, I changed the includes HTML to be as simple as possible, by directly passing in the _full URL, with our without the ID kwargs_, instead of the caluclating the `url` from the `route` in the template. I added a new context key/value, `previous_url`, which is now being set in `views.py`.

Comment (angela-tran) [reply web/vital_records/templates/vital_records/_button_row.html:4]: That makes sense, nice fix!

For the views that are inheriting from `StepsMixin`, can we move
```
context["previous_url"] = reverse(context["previous_route"], kwargs={"pk": self.object.pk})
```
(e.g. in [`NameView`](https://github.com/Office-of-Digital-Services/cdt-ods-disaster-recovery/pull/319/files#diff-5f9001db61faca1d31575b962487dec922eae2664a66f78cfb1e0f8695a8e929R139)) into the mixin?

**Comment by machikoyasuda** [comment]
I'm starting to think maybe we should wait for all the form/order pages to be finished, then we revisit this PR for review and test. That way, we can have all flow pages in, and test the back button on the full Birth and Marriage flows. Thoughts? @angela-tran @lalver1

**Comment by angela-tran** [comment]
> I'm starting to think maybe we should wait for all the form/order pages to be finished, then we revisit this PR for review and test. That way, we can have all flow pages in, and test the back button on the full Birth and Marriage flows. Thoughts? @angela-tran @lalver1

Sure, that's fine with me

**Comment by machikoyasuda** [comment]
_Ohkay_, now it's really ready for review. I've been doing lots of birth flows, then marriage flow, and going back and forth between all the pages. @angela-tran @lalver1

File: web/vital_records/views/common.py [hunk 3]
```diff
@@ -143,6 +145,7 @@ def form_valid(self, form):
 class CountyView(StepsMixin, EligibilityMixin, ValidateRequestIdMixin, UpdateView):
     model = VitalRecordsRequest
     template_name = "vital_records/request/form.html"
+    step_name = Steps.county_of_birth
```
Comment (angela-tran) [inline web/vital_records/views/common.py:148]: This looks like it might've snuck back in during a rebase

File: web/vital_records/views/common.py [hunk 4]
```diff
@@ -86,7 +86,8 @@ class TypeView(EligibilityMixin, ValidateRequestIdMixin, UpdateView):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["page_title"] = "Replacement records"
-        context["previous_route"] = Routes.request_start
+        context["previous_route"] = Routes.app_route(Routes.request_start)
+        context["previous_url"] = reverse(context["previous_route"])
```
Comment (angela-tran) [inline web/vital_records/views/common.py:90]: This is nitpicky, but... since the template no longer uses `previous_route`, we don't need it to be in the `context` dictionary, right?

We could have it as just a local variable, like:
```suggestion
        previous_route = Routes.app_route(Routes.request_start)
        context["previous_url"] = reverse(previous_route)
```

Same comment for all the other views and the mixin where we say `context["previous_url"] = reverse(...`

**Review by angela-tran — CHANGES_REQUESTED** [review]
This is looking good @machikoyasuda . I just noticed some small clean-up to do

**Comment by angela-tran** [comment]
The spacing around the "Confirm" button on the `start.html` page seems to have changed. Do you see this too @machikoyasuda?

| `main` | this branch |
| --- | --- |
|image | image |

**Comment by machikoyasuda** [comment]
@angela-tran I am not seeing that start page spacing change locally. No HTML has changed on that page.

**Review by angela-tran — APPROVED** [review]
Nice work @machikoyasuda!
